### PR TITLE
fix: make sure Bun.sleep(Date) doesn't resolve prematurely

### LINKED ISSUE #8834: Bun.sleep() rounds incorrectly
### What version of Bun is running?

1.0.26+c75e768a6

### What platform is your computer?

Linux 6.1.31-gentoo-dist x86_64 13th Gen Intel(R) Core(TM) i7-1370P

### What steps can reproduce the bug?

Run this code (based on a test from setTimeout.test.js):

```
while (true) {
    let ten_ms = new Date();
    ten_ms.setMilliseconds(ten_ms.getMilliseconds() + 12);
    await Bun.sleep(ten_ms);
    let now = new Date();
    if (!(+ten_ms <= +now)) {
	console.error({ten_ms: +ten_ms, now: +now});
	break;
    }
}
```

### What is the expected behavior?

An infinite loop.

### What do you see instead?

A break, since eventually Bun.sleep() will return a millisecond before the specified instant has been reached.

### Additional information

The fix is simply adding:

```
    millisecondsValue = JSC::jsNumber(ceil(millisecondsValue.asNumber()));
```

in BunObject.cpp:functionBunSleep. This will also make Bun.sleep(.001) sleep for a millisecond, which seems like safe behavior.

See #8833.

### PR DESCRIPTION
**Pull request by argosphil**
Fixes #8834.

Replaces #8833, #8859.

This makes Bun.sleep(new Date(number)) fulfill its promise only when Date.now() >= number.

### What does this PR do?

This is a very limited PR which adds a single line of code (plus tests), to ensure that we sleep at least as long as the difference between the current date and the target date. The previous code rounded towards zero in some circumstances, resulting in sleeps that were too short.

- [x] Code changes

### How did you verify your code works?

I wrote automated tests

### CODE REVIEW

File: test/js/web/timers/setTimeout.test.js [hunk 1]
```diff
@@ -157,6 +157,14 @@ it("Bun.sleep works with a Date object", async () => {
   expect(performance.now() - now).toBeGreaterThan(10);
```
Comment (paperclover) [inline test/js/web/timers/setTimeout.test.js]: <i class='graphite__hidden'>[Re: line 155]<br/><br/></i>

can you resolve this test?<span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/oven-sh/bun/8950?utm_source=unchanged-line-comment">Graphite</a>.</span>

**Review by paperclover — CHANGES_REQUESTED** [review]
i think this is much better than modifying setTimeout.

can you resolve the todo i marked earlier about this issue?

**Review by paperclover — APPROVED** [review]
great

**Comment by jdalton** [comment]
Thank you @argosphil!

**Review by jdalton — APPROVED** [review]

**Pull request merged**